### PR TITLE
More locks

### PR DESCRIPTION
**Pull request by LMESTM**
@jeromecoutant 
Untested patch as a proposal ... to be reviewed / tested

### CODE REVIEW

**Comment by jeromecoutant** [comment]
@JeanMarcR do you have any advice ?

File: ISM43362Interface.cpp [hunk 1]
```diff
@@ -443,14 +444,20 @@ int ISM43362Interface::socket_send_nolock(void *handle, const void *data, unsign
...
+int ISM43362Interface::socket_recv_nolock(void *handle, void *data, unsigned size)
+{
     unsigned recv = 0;
     struct ISM43362_socket *socket = (struct ISM43362_socket *)handle;
     char *ptr = (char *)data;
 
     // debug_if(_ism_debug, "ISM43362Interface socket_recv: req=%d read_data_size=%d connected %d\r\n", size, socket->read_data_size, socket->connected);
 
     if (!socket->connected) {
-        _mutex.unlock();
```
Comment (JanneKiiskila) [inline ISM43362Interface.cpp:453]: You probably can't remove this? The lock is acquired in the beginning of this function.

**Comment by jeromecoutant** [comment]
@LMESTM maybe this one is replaced by #90 ?

**Comment by LMESTM** [comment]
yes definitely !  I'm closing this one - superseded by #90

**Pull request closed without merging**